src/engine/SmtState.h

```diff
@@ -0,0 +1,46 @@
+/*********************                                                        */
+/*! \file SmtState.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Guy Katz, Duligur Ibeling
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+
+ **/
+
+#ifndef __SmtState_h__
+#define __SmtState_h__
+
+#include "List.h"
+#include "Map.h"
+#include "PiecewiseLinearConstraint.h"
+#include "StackEntry.h"
+
+class SmtState
```

Comment:
Looking at the current SmtCore, it also has a _stateId field. Shouldn't this be part of the SmtState?

Reply:
Added the stateId as part of the SmtState